hook.sh

```diff
@@ -46,17 +46,40 @@ echo -e "\e[32mSetting challenge to ${CERTBOT_VALIDATION} ...\e[0m"
 domain=.$CERTBOT_DOMAIN
 infix=${domain%.$DEDYN_NAME}
 
+# Remove leading wildcard from infix, if present
+# *.foobar.dedyn.io gives "" while *.a.foobar.dedyn.io gives ".a"
+infix=${infix#.\*}
+
 args=( \
-    '-Ss' \
+    '-sSLf' \
     '-H' "Authorization: Token $DEDYN_TOKEN" \
     '-H' 'Accept: application/json' \
     '-H' 'Content-Type: application/json' \
-    '-d' '[{"subname":"_acme-challenge'"$infix"'", "type":"TXT", "records":["\"'"$CERTBOT_VALIDATION"'\""], "ttl":60}]' \
-    '-o' '/dev/null' \
 )
 
+# Letsencrypt uses a TTL of 60 seconds on their resolvers. If we're getting
+# wildcard certificates (e.g. foobar.dedyn.io and *.foobar.dedyn.io), multiple
+# certificates for the same domain (e.g. RSA and ECDSA), or both, we'll need to
+# wait 60 seconds for the records to expire before verification begins. We'll
+# store any existing challenges and publish them along with the current one.
+# Set timeout_ttl=yes in ./dedynauth if you want to enable this.
+# You'll probably also want to use the optional cleanup hook to keep records
+# from accumulating.
+
+readarray -t acme_live <<< "$(curl "${args[@]}" -X GET "https://desec.io/api/v1/domains/$DEDYN_NAME/rrsets/_acme-challenge$infix/TXT" \
+    | grep -o '"records":\[[^]]*]' | grep -o '\\"[^,]*\\"')"
+
+[ ${#acme_live[@]} -gt 0 ] && [ -n "${acme_live[*]}" ] && [ -n "$timeout_ttl" ] && \
```

Comment:
- Is it even possible that the first and the second condition are not true at the same time?
- The last condition is also true if you set `timeout_ttl=no`. The comment further up implies that `sleep` is only called if the values is `yes`. We should either be checking that here, or change the comment to say `Set timeout_ttl to any value ...`.

Reply:
Yes, but only because a null string in an array counts as an entry. Will be avoided by the changes above.

I agree with the timeout ambiguity. I'm going to make the comments a little more general, and the comparison more explicit.